src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadata.cs

```diff
@@ -528,5 +529,35 @@ public override IReadOnlyList<object> ValidatorMetadata
                 return _details.PropertySetter;
             }
         }
+
+        public override ValidationErrorMessages ValidationErrorMessages
+        {
+            get
+            {
+                if (_validationErrorMessages == null)
+                {
+                    _validationErrorMessages = new DefaultErrorMessages(ValidationMetadata.ValidationErrorMetadata);
```

Comment:
Yeah I think your suggested change is a good choice if you decide to make a defensive copy inside of `DefaultValidationMetadataProvider`.

Due to how `new` and `abstract` interact, I think you'll need 3 classes.